source/SyntheticTopology/SierpinskiObject.lagda

```diff
@@ -0,0 +1,155 @@
+open import UF.PropTrunc
+open import UF.Sets
+open import UF.Sets-Properties
+open import UF.Subsingletons
+open import UF.SubtypeClassifier
+
+module SyntheticTopology.SierpinskiObject
+        (fe : Fun-Ext)
+        (pe : Prop-Ext)
+        (pt : propositional-truncations-exist) where
+
+open import Dominance.Definition
+open import UF.Classifiers
+open import UF.Embeddings
+open import UF.Equiv
+open import UF.ImageAndSurjection pt
+open import UF.Logic
+open import UF.Subsingletons-FunExt
+open import UF.Univalence
+open import UF.UniverseEmbedding
+
+open AllCombinators pt fe
+open PropositionalTruncation pt hiding (_∨_)
+
+\end{code}
+
+What is a Sierpiński object? In Martín Escardó´s unpublished note [_Topology via
+higher-order intuitionistic logic_][1], a Sierpiński object is defined in the
+setting of a topos as a subobject of the subobject classifier. This is also
+given in Definition 2.4 of Davorin Lesnik's thesis, who took this unpublished
+note as a starting point for his PhD thesis.
+
+The purpose of this development is to develop these notions in the context of
+HoTT/UF, where we look at subtypes of the subtype classifier. Because we work
+predicatively, however, the definition of the notion of Sierpiński object is not
+that straightforward in our setting.
+
+In the impredicative setting of topos theory, one works with the topos of 𝓤-sets
+over some universe 𝓤, and the Sierpiński object in that setting would translate
+to:
+
+\begin{code}
+
+Sierpinski-Object₀ : (𝓤 : Universe) → 𝓤 ⁺  ̇
+Sierpinski-Object₀ 𝓤 = Subtype' 𝓤 (Ω 𝓤)
+
+\end{code}
+
+However, many dominances that we are interested in do not fit this definition in
+a predicative setting. For example, the largest possible dominance of `Ω 𝓤` is
+𝓤⁺-set rather than a 𝓤-set. Accordingly, we generalize the universes in the
+notion of Sierpiński object as follows:
+
+\begin{code}
+
+Generalized-Sierpinski-Object : (𝓤 𝓥 : Universe) → (𝓤 ⊔ 𝓥) ⁺  ̇
+Generalized-Sierpinski-Object 𝓤 𝓥 = Ω 𝓤 → Ω 𝓥
+
+\end{code}
+
+We think that in most cases, our dominances will be of the form `Ω 𝓤 → Ω (𝓤 ⁺)`,
+but because we can prove things in this general setting, we choose to work with
+the generalized definition instead.
+
+We define some notation to refer to components of a Sierpiński object.
+
+In the module below, we assume the existence of a Sierpiński object `𝕊` and
+define some notions _synthetically_, following the work of Martín Escardó (and
+Davorin Lešnik).
+
+\begin{code}
+
+module Sierpinski-notations (𝕊 : Generalized-Sierpinski-Object 𝓤 𝓥) where
+
+\end{code}
+
+The propositions in `Ω` that fall in the subset `𝕊` are called _open
+propositions_. We introduce suggestive terminology accordingly.
+
+\begin{code}
+
+ is-open-proposition : Ω 𝓤 → Ω 𝓥
+ is-open-proposition = 𝕊
+
+\end{code}
+
+Here, we only work with sets.
+
+To define this and some related notions, we work in a module parameterized by an
+hSet `𝒳`. We adopt the convention of using calligraphic letters `𝒳`, `𝒴`, ...
```

Comment:
We should either stick to this convention in other modules or remove this remark from this module.

Reply:
@supermartruc Has this been fixed everywhere now?

Reply:
@martintrucchi: Can I mark this as resolved now?

Reply:
Yes, I used it everywhere